proxy/src/outbound.rs

```diff
@@ -51,81 +51,78 @@ impl<B> Outbound<B> {
 }
 
 #[derive(Clone, Debug, PartialEq, Eq, Hash)]
-pub enum Destination {
-    LocalSvc(FullyQualifiedAuthority),
-    External(SocketAddr),
-}
-
-impl From<FullyQualifiedAuthority> for Destination {
-    #[inline]
-    fn from(authority: FullyQualifiedAuthority) -> Self {
-        Destination::LocalSvc(authority)
-    }
+pub enum Dst {
+    Unbound(FullyQualifiedAuthority, Protocol),
+    Bound(SocketAddr, Protocol),
 }
 
-impl From<SocketAddr> for Destination {
-    #[inline]
-    fn from(addr: SocketAddr) -> Self {
-        Destination::External(addr)
-    }
-}
-
-
 impl<B> Recognize for Outbound<B>
 where
     B: tower_h2::Body + 'static,
 {
     type Request = http::Request<B>;
     type Response = bind::HttpResponse;
     type Error = <Self::Service as tower::Service>::Error;
-    type Key = (Destination, Protocol);
+    type Key = Dst;
     type RouteError = bind::BufferSpawnError;
     type Service = InFlightLimit<Timeout<Buffer<Balance<
         load::WithPendingRequests<Discovery<B>>,
         choose::PowerOfTwoChoices<rand::ThreadRng>
     >>>>;
 
+    /// Determines the destination service for a given Request.
+    ///
+    /// Destinations are either bound or unbound. Unbound destinations must be resolved
+    /// through the controller to produce a list of endpoints. Bound destinations use the
+    /// client-provided destination address.
+    ///
+    /// For HTTP/2 requests, a service may support several :authority values. HTTP/1
+    /// requests, on the other hand, may not share a service across authorities.
```

Comment:
I suggest: "HTTP/2 connections may multiplex connections for multiple authorities (as identified by `:authority`) on the same connection HTTP/1 connections must have one authority per connection." In particular, this isn't a per-service thing, but a per-connection thing. Also, please move this comment to be with the code that implements this logic. (See my note below.)

Reply:
I think it actually _is_ about services and not connections:

the router contract is that Recognize produces a key that is used to bind a service. this service may maintain an arbitrary number of connections.  we care that we have a service for each unique authority.